# HG changeset patch
# User Michael C. Grant <mcg@cvxr.com>
# Date 1390516663 28800
#      Thu Jan 23 14:37:43 2014 -0800
# Branch stable
# Node ID c460566106f07dc809b5a0f23663867b2e4c1b41
# Parent  50505bf88b8257300ab590c72f4245af14a4a7b1
configure.ac: Find Qt libs in framework option for Macs (bug #41236)

* configure.ac: On Darwin OS only, look for Qt libraries in Framework
directory.

* src/Makefile.am: Add missing line continuation to octave_cli_LDADD
definition list.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -178,23 +178,23 @@ AC_ARG_WITH([sepchar],
   [AS_HELP_STRING([--with-sepchar=<char>],
     [use <char> as the path separation character])])
 case $with_sepchar in
   yes | "")
     case $host_os in
       mingw* | msdosmsvc)
         sepchar=';' ;;
     esac
-    ;;
+  ;;
   no)
     AC_MSG_ERROR([You are required to define a path separation character])
     ;;
   *)
     sepchar=$with_sepchar
-    ;;
+  ;;
 esac
 AC_SUBST(sepchar)
 AC_DEFINE_UNQUOTED(SEPCHAR, ['$sepchar'],
   [Define this to be the path separator for your system, as a character constant.])
 AC_DEFINE_UNQUOTED(SEPCHAR_STR, ["$sepchar"],
   [Define this to be the path separator for your system, as a string.])
 
 ### Define the path to the shell on the host system.  Most systems will
@@ -2299,17 +2299,17 @@ case $host_os in
   ;;
 esac
 
 ## Windows-specific use of functions
 case $host_os in
   msdosmsvc | mingw*)
     AC_CHECK_FUNCS([setvbuf], [],
                    [AC_MSG_ERROR([Missing function required to build Octave])])
-    ;;
+  ;;
 esac
 
 ## Cygwin kluge for getrusage.
 AC_CHECK_FUNCS([getrusage])
 case $host_os in
   cygwin*)
     AC_DEFINE(RUSAGE_TIMES_ONLY, 1,
       [Define to 1 if your struct rusage only has time information.])
@@ -2718,16 +2718,25 @@ if test $build_gui = yes; then
      build_gui=no])
 
   if test $build_gui = yes; then
     ## Retrieve Qt compilation and linker flags
     QT_CPPFLAGS="`$PKG_CONFIG --cflags-only-I QtCore QtGui QtNetwork`"
     QT_LDFLAGS="`$PKG_CONFIG --libs-only-L QtCore QtGui QtNetwork`"
     QT_LIBS="`$PKG_CONFIG --libs-only-l QtCore QtGui QtNetwork`"
 
+    case $host_os in
+      *darwin*)
+        ## Qt might be installed in framework
+        QT_LDFLAGS="`$PKG_CONFIG --libs-only-other QtCore QtGui QtNetwork | tr ' ' '\n' | $GREP -e '-F' | uniq | tr '\n' ' '`"
+        QT_LIBS="`$PKG_CONFIG --libs-only-other QtCore QtGui QtNetwork | tr ' ' '\n' | $GREP -v -e '-F' | uniq | tr '\n' ' '`"
+        AM_CONDITIONAL([AMCOND_LINK_ALL_DEPS], [test $link_all_deps = yes || test -n $QT_LDFLAGS])
+      ;;
+    esac
+
     ## Check for Qt4
     if ! `$PKG_CONFIG --atleast-version=4.0.0 QtCore`; then
       warn_gui="Qt >= 4.0.0 not found -- disabling GUI"
       build_gui=no
     fi
   fi
 
   if test $build_gui = yes; then
@@ -2804,25 +2813,25 @@ if test $build_gui = yes; then
        build_gui=no])
 
     if test $build_gui = yes; then
       case $host_os in
         mingw* | msdosmsvc*)
           AC_CHECK_FUNCS([setvbuf], [win32_terminal=yes],
             [warn_gui="setvbuf not found -- disabling GUI"
              build_gui=no])
-          ;;
+        ;;
         *)
           AC_CHECK_HEADERS([pty.h libutil.h util.h])
           AC_SEARCH_LIBS([openpty], [util],
             [AC_DEFINE(HAVE_OPENPTY, [], [Define whether openpty exists])])
           AC_CHECK_FUNCS([chmod chown ftruncate mmap munmap], [],
             [AC_MSG_ERROR([At least one of chmod, chown, ftruncate, mmap, and munmap not found -- disabling GUI])
              build_gui=no])
-          ;;
+        ;;
       esac
     fi
   fi
 
   if test -n "$warn_gui"; then
     OCTAVE_CONFIGURE_WARNING([warn_gui])
   fi
 fi
diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -82,17 +82,17 @@ endif
 
 octave_CPPFLAGS = \
   $(AM_CPPFLAGS) \
   $(OCTAVE_CPPFLAGS)
 
 octave_cli_SOURCES = main-cli.cc
 
 octave_cli_LDADD = \
-  $(OCTAVE_CORE_LIBS)
+  $(OCTAVE_CORE_LIBS) \
   $(OCTAVE_LINK_DEPS)
 
 octave_cli_LDFLAGS = \
   $(NO_UNDEFINED_LDFLAG) \
   $(OCTAVE_LINK_OPTS)
 
 if AMCOND_BUILD_GUI
   octave_gui_SOURCES = main-gui.cc
